### PR TITLE
[DOCS] Split most of README.md to GettingStarted.md

### PR DESCRIPTION
**Pull request by xgupta**
README text should remain minimal in a project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,10 +5,8 @@
 
 # ⚡️ "CIRCT" / Circuit IR Compilers and Tools
 
-"CIRCT" stands for "Circuit IR Compilers and Tools".  One might also interpret
```
Comment (lattner) [inline README.md:8]: I personally am a fan of the cheeky intro, but I think that explicitly mentioning MLIR in the first paragraph is a good idea!

File: README.md [hunk 2]
```diff
@@ -23,38 +21,11 @@ for more information.
    requests for the CIRCT repository, and gain commit access using the [standard
    LLVM policies](https://llvm.discourse.group/c/Projects-that-want-to-become-official-LLVM-Projects/circt/).
 
-Also take a look at the following diagram, which gives a brief overview of the current [dialects and how they interact](docs/includes/img/dialects.svg):
-
-<p align="center"><img src="docs/includes/img/dialects.svg"/></p>
-
-## Motivation
```
Comment (lattner) [inline README.md:30]: I'm not sure that dropping this adds a lot of value - a motivation section in the getting started guide is a bit weird.  @stephenneuendorffer WDYT about this proposal?

Comment (xgupta) [reply README.md:30]: Yeah, correct. The right place of motivation should be in the charter doc.

Comment (xgupta) [reply README.md:30]: Not sure we should update the charter doc once committed? Otherwise, I will just leave the motivation text in README.md.

Comment (teqdruid) [reply README.md:30]: IIRC, the charter is (or perhaps intended to be) essentially a longer version of this motivation. I would keep the motivation in the README.

**Comment by xgupta** [comment]
Thanks, @lattner, @teqdruid for reviewing. I update docs according to comments. Please take another look.

**Review by lattner — APPROVED** [review]
This revision LGTM, thanks!  I'd still like @stephenneuendorffer and @teqdruid to approve as well though.

**Review by teqdruid — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**